uint8 NONE = 0
uint8 FLOAT = 1
uint8 FIXED = 2
uint8 NO_SOL = 3

uint8 fix